samples/media/ultrahdr/src/main/java/com/example/platform/media/ultrahdr/video/UltraHDRToHDRVideo.kt

```diff
@@ -0,0 +1,629 @@
+     */
+    private lateinit var muxer: MediaMuxer
+
+    /**
+     * [MediaMuxer] track id, to identify which track is the encoding track.
+     */
+    private var muxerTrackId = -1
+
+    /**
+     * Folder path for the .mp4 files to be generated too
+     */
+    private val folderPath: String by lazy {
+        requireActivity().cacheDir.path + '/' + FOLDER_PATH_NAME + '/'
+    }
+
+    /**
+     * file path for the .mp4 file to be generated too
+     */
+    private val filePath: String by lazy {
+        folderPath + UUID.randomUUID().toString() + VIDEO_FILE_EXTENSION
+    }
+
+    /**
+     * [MediaCodec] encoder that will be used to encode the the HDR video.
+     */
+    private lateinit var encoder: MediaCodec
+
+    /**
+     * The encoders [Handler] & [HandlerThread], used to process the [MediaCodec.Callback], which
+     * process frames from the [ImageWriter]
+     */
+    private lateinit var encoderHandler: Handler
+    private val encoderThread: HandlerThread by lazy {
+        HandlerThread("encoderThread")
+    }
+
+    /**
+     * [Surface], produced from [MediaCodec] that will, that is used by the [ImageWriter] to fill
+     * with image data to be encoded by the encoder.
+     */
+    private lateinit var encoderSurface: Surface
+
+    /**
+     * Flag to determine whether or not the frame has been processed by the [MediaCodec] & written
+     * to the [MediaMuxer].
+     *
+     * Using [AtomicBoolean] since it can be accessed by multiple threads.
+     */
+    private var isFrameBufferProcessed = AtomicBoolean(true)
+    private var framesProcessed = AtomicInteger(0)
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, s: Bundle?): View {
+        _binding = UltrahdrToHdrVideoBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // Check that hw acceleration is supported
+        if (!isHardwareAccelerationSupported()) return
+        if (!setUpMediaCodec()) return
+
+        // Setup Image Reader.
+        setUpImageWriter()
+
+        // Attaching ColorMode Controller for Activity HDR mode for images
+        binding.colorModeControls.setWindow(requireActivity().window)
+
+        // Load UltraHDR Preview Images
+        loadImageIntoView(ULTRA_HDR_IMAGES[0], binding.imageContainer1)
+        loadImageIntoView(ULTRA_HDR_IMAGES[1], binding.imageContainer2)
+        loadImageIntoView(ULTRA_HDR_IMAGES[2], binding.imageContainer3)
+
+        binding.covertButton.setOnClickListener {
+            binding.covertButton.isEnabled = false
+
+            // Perform conversion task on the IO Dispatcher
+            CoroutineScope(Dispatchers.IO).launch {
+                val threadName = Thread.currentThread().name
+                Log.i(TAG, "Starting Conversion to Video on $threadName")
+                doImageToVideoConversion()
```

Comment:
This coroutine is launched on the `IO` `Dispatcher`, and subsequently used `withContext` to also place it on the same dispatcher. The latter alone should be enough.